src/chains/ethereum/block/src/runtime-block.ts

```diff
@@ -163,10 +166,14 @@ export class RuntimeBlock {
       BUFFER_8_ZERO // nonce
     ];
     const { totalDifficulty } = header;
-    const txs: EthereumRawTx[] = [];
+    const txs: BlockDatabaseTransaction[] = []; // we don't want
     const extraTxs: GanacheRawBlockTransactionMetaData[] = [];
     transactions.forEach(tx => {
-      txs.push(tx.raw);
+      if (tx.raw[0][0] === undefined) {
```

Comment:
Use `TransactionFactory.typeOfRaw(tx.raw) === LegacyTransaction` instead. There are some edge cases here where a LegacyTransaction could be mistaken for a typed one.